scripts/new-version-pr-generator.js

```diff
@@ -0,0 +1,180 @@
+/* eslint-disable no-console */
+
+const {execSync} = require('child_process');
+const {readFileSync} = require('fs');
+const pathResolve = require('path').resolve;
+const {mkdir, rm} = require('shelljs');
+const yaml = require('js-yaml');
+const semver = require('semver');
+
+const {version: PACKAGE_VERSION} = require('../package.json');
+const secrets = require('../secrets.json');
+const Retry = require('./utilities/retry');
+
+const YARN_VERSION = yaml
+  .safeLoad(readFileSync(pathResolve(__dirname, '..', 'dev.yml'), 'utf8'))
+  .up.find((obj) => Object.keys(obj).includes('node')).node.yarn;
+
+// File paths and directories
+const root = pathResolve(__dirname, '../');
+const sandbox = pathResolve(root, 'sandbox');
+
+// Create the sandbox directory
+rm('-rf', sandbox);
+mkdir(sandbox);
+
+// Current release version
+const releaseVersion = `v${PACKAGE_VERSION}`;
+
+// Compute the base branch on the repository (default: master)
+// Example: will open a PR against the v4 branch if the
+// version found in package.json is `4.0.0-beta.x`
+
+// Check if a version is a major pre-release:
+// isMajorPrerelease('2.0.0') => false
+// isMajorPrerelease('3.0.1-beta') => false
+// isMajorPrerelease('1.1.0-alpha') => false
+// isMajorPrerelease('2.0.0-beta.1') => true
+// isMajorPrerelease('3.0.0-alpha') => true
+function isMajorPrerelease(version) {
+  return (
+    semver.prerelease(version) &&
+    semver.minor(version) === 0 &&
+    semver.patch(version) === 0
+  );
+}
+
+const baseBranch = isMajorPrerelease(PACKAGE_VERSION)
+  ? `v${semver.major(PACKAGE_VERSION)}`
+  : 'master';
+
+// Polaris bot settings
+const polarisBotName = 'Shopify Polaris Bot';
+const polarisBotEmail = 'shopify-polaris-bot@users.noreply.github.com';
+const polarisBotToken = secrets.github['shopify-polaris'];
+const shopifyPolarisBotGitOverride = `GIT_COMMITTER_NAME='${polarisBotName}' GIT_COMMITTER_EMAIL='${polarisBotEmail}'`;
+
+/**
+ * BulkExecute - Run a bunch of execSync one at a time
+ *
+ * @param {array}  tasks   - List of tasks to run
+ * @param {object} options - Specific options for execSync
+ */
+function BulkExecute(tasks, options) {
```

Comment:
I don't think this function is necessary. You're running it once so it doesn't save you any code, and it's simple enough that it doesn't benefit from being encapsulated either. What do you think of removing it?

Reply:
Agreed!